designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/DeploymentManager.scala

```diff
@@ -60,11 +60,12 @@ trait DeploymentManager extends AutoCloseable {
 
   def cancel(name: ProcessName, deploymentId: DeploymentId, user: User): Future[Unit]
 
-  def test(
+  def test[T](
       name: ProcessName,
       canonicalProcess: CanonicalProcess,
-      scenarioTestData: ScenarioTestData
-  ): Future[TestResults]
+      scenarioTestData: ScenarioTestData,
+      variableEncoder: Any => T // NU-1455: We encode variable on the engine, because of classLoader's problems
```

Comment:
IMO we could write more here about this problem - let's write that class `AvroToJsonEncoder` wasn't used, compared classes had other classloaders, which class was in the stack trace etc. I think that it is a material for the method level scaladoc